Framework/DataHandling/test/ReadMaterialTest.h

```diff
@@ -0,0 +1,293 @@
+    const ReadMaterial::MaterialParameters params = [this]() -> auto {
+      ReadMaterial::MaterialParameters setMaterial;
+      setMaterial.chemicalSymbol = FORMULA;
+      setMaterial.sampleNumberDensity = 1;
+      setMaterial.coherentXSection = 1;
+      setMaterial.incoherentXSection = 2;
+      setMaterial.attenuationXSection = 3;
+      setMaterial.scatteringXSection = 4;
+      return setMaterial;
+    }
+    ();
+
+    ReadMaterial reader;
+    reader.setMaterialParameters(params);
+    auto material = reader.buildMaterial();
+
+    Kernel::MaterialBuilder builder;
+    builder.setFormula(FORMULA);
+    builder.setNumberDensity(1);
+    builder.setCoherentXSection(1);
+    builder.setIncoherentXSection(2);
+    builder.setAbsorptionXSection(3);
+    builder.setTotalScatterXSection(4);
+    auto check = builder.build();
+
+    compareMaterial(material, check);
+  }
+
+  void testGenerateScatteringInfo() {
+    const ReadMaterial::MaterialParameters params = [this]() -> auto {
+      ReadMaterial::MaterialParameters setMaterial;
+      setMaterial.chemicalSymbol = FORMULA;
+      setMaterial.sampleNumberDensity = 1;
+      return setMaterial;
+    }
+    ();
+
+    ReadMaterial reader;
+    reader.setMaterialParameters(params);
+    auto material = reader.buildMaterial();
+
+    Kernel::MaterialBuilder builder;
+    builder.setFormula(FORMULA);
+    builder.setNumberDensity(1);
+    builder.setCoherentXSection(0.0184000000);
+    builder.setIncoherentXSection(5.0800000022);
+    builder.setAbsorptionXSection(5.0800000022);
+    builder.setTotalScatterXSection(5.1000000044);
+    auto check = builder.build();
+
+    compareMaterial(material, check);
+  }
+
+  void testNoMaterialFailure() {
+    const ReadMaterial::MaterialParameters params = [this]() -> auto {
+      ReadMaterial::MaterialParameters setMaterial;
+      setMaterial.chemicalSymbol = EMPTY;
+      setMaterial.atomicNumber = 0;
+      setMaterial.massNumber = 0;
+      setMaterial.sampleNumberDensity = 1;
+      setMaterial.zParameter = EMPTY_DOUBLE_VAL;
+      setMaterial.unitCellVolume = EMPTY_DOUBLE_VAL;
+      setMaterial.sampleMassDensity = EMPTY_DOUBLE_VAL;
+      return setMaterial;
+    }
+    ();
+
+    ReadMaterial reader;
+    reader.setMaterialParameters(params);
+    TS_ASSERT_THROWS(reader.buildMaterial(), std::runtime_error);
+  }
+
+private:
+  const double EMPTY_DOUBLE_VAL = 8.9884656743115785e+307;
+  const std::string EMPTY = "";
+  const std::string FORMULA = "V";
+
+  void compareMaterial(const std::unique_ptr<Material> &material,
+                       const Material &check) {
+    std::vector<Material::FormulaUnit> checkFormula = check.chemicalFormula();
+    std::vector<Material::FormulaUnit> materialFormula =
+        material->chemicalFormula();
+
+    TS_ASSERT_EQUALS(material->numberDensity(), check.numberDensity());
+    TS_ASSERT_DELTA(material->cohScatterXSection(), check.cohScatterXSection(),
+                    0.00000001);
```

Comment:
`1e-8` is  a nicer way to do this. Also perhaps use a constant instead o a magic number?